### PR TITLE
Added --check-unextracted to extract mix task

### PR DESCRIPTION
**Pull request by akoutmos**
This PR addresses #286.

While still a work in progress (mostly test clean up that I want to do), I figured I would throw up the PR to get eyes on it sooner rather than later in case the approach I took was off.

Thanks in advance for reviewing the PR :)

### CODE REVIEW

**Comment by juanperi** [comment]
hey, this would be an awesome addition!
I have a question though. From the look at the PR, you are exactly matching the content of the previous file vs the new file, and failing if they don't.
Now, that is indeed correct, and very helpful. But what if there are no new translations, and the only thing that changed is the line in which a translation was taken from.
eg:
```
#, elixir-format
#: lib/app_web/templates/compare/index/_alternative_card.html.eex:49
msgid "# 1 Alternative"
msgstr ""
```
If we add a new line somewhere in that template, now the line will be 50, and your check will fail.

So, when should this test fail?:
1. If a new `gettext("content")` was added and it was not extracted
2. if any possible change (like line change) has happened and would make the POT file different

**Comment by josevalim** [comment]
@juanperi I think the goal is still raise, because it means someone forgot to run the command. My understanding is that the goal of the flag is basically "did someone remember to run mix getttext.extract before committing?". :) But I am interested in @akoutmos' take too.

**Comment by akoutmos** [comment]
I thought about this as well yesterday...and the conclusion that I came to was that while there are no new translations and the existing translations either changed lines/files, the POT file is still inaccurate with regards to the current state of the codebase. Hence why I made the PR the way that I did.

**Comment by josevalim** [comment]
Hrm, maybe we should change the name? After all it is checking more than unextracted. Maybe `--check-up-to-date` or `--check-updated`?

**Comment by akoutmos** [comment]
I guess `--check-unextracted` might be a bit misleading now that I think about it. I'd be ok with `--check-up-to-date` or perhaps `--check-current` for something shorter?

**Comment by josevalim** [comment]
Any is fine. I am glad to leave it up to you. :)

**Comment by juanperi** [comment]
hey, awesome we reached a consensus on the goal of the check. Thank you both.

I don't think that i would be able to use it for my particular case, where most of the time changes in the templates do not imply they are adding new translations, and although the line numbers are useful, they are mostly needed when new translations are made available. But I understand that strictness is better for a general check :)

Maybe the extra "configuration" flag for this check might be implemented later if you think it would be useful (i might work on that if there is interest)

**Comment by akoutmos** [comment]
Finished the last bit of test clean up so all set on my end :).

**Review by josevalim — APPROVED** [review]

File: lib/mix/tasks/gettext.extract.ex [hunk 1]
```diff
@@ -14,6 +14,13 @@ defmodule Mix.Tasks.Gettext.Extract do
   and `:priv` options given by Gettext modules when they call `use Gettext`. One
   POT file is generated for each translation domain.
 
+  If you would like to verify that your POT files are up to date with the
+  current state of the codebase, you can provide the `--check-up-to-date`
+  flag. Note that this validation will fail even when the same calls to gettext
+  only change location in the codebase:
+
+      mix gettext.extract --check-up-to-date
```
Comment (whatyouhide) [inline lib/mix/tasks/gettext.extract.ex:22]: Love the flag name. Awesome 💟

**Pull request merged**